List[index]::toOptionalStringRef

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42263 List[index]::toOptionalStringRef**

Allow a way to get a reference to the stored string in a `List<optional<string>>` without having to copy the string.
This for example improves perf of the map_lookup op by 3x.

Differential Revision: [D22830381](https://our.internmc.facebook.com/intern/diff/D22830381/)

smessmer committed:
List[index]::toOptionalStringRef

Allow a way to get a reference to the stored string in a `List<optional<string>>` without having to copy the string.
This for example improves perf of the map_lookup op by 3x.

Differential Revision: [D22830381](https://our.internmc.facebook.com/intern/diff/D22830381/)

[ghstack-poisoned]

smessmer committed:
Update on "List[index]::toOptionalStringRef"


Allow a way to get a reference to the stored string in a `List<optional<string>>` without having to copy the string.
This for example improves perf of the map_lookup op by 3x.

Differential Revision: [D22830381](https://our.internmc.facebook.com/intern/diff/D22830381/)

[ghstack-poisoned]

ezyang reviewed (approved):
A bit smelly that we have to do this for every single optional type like this :/

smessmer commented:
> A bit smelly that we have to do this for every single optional type like this :/

We can only do it for optional types that are directly stored in the IValue. It is not possible for Tensor for example.

smessmer committed:
Update on "List[index]::toOptionalStringRef"


Allow a way to get a reference to the stored string in a `List<optional<string>>` without having to copy the string.
This for example improves perf of the map_lookup op by 3x.

Differential Revision: [D22830381](https://our.internmc.facebook.com/intern/diff/D22830381/)

[ghstack-poisoned]

smessmer committed:
Update on "List[index]::toOptionalStringRef"


Allow a way to get a reference to the stored string in a `List<optional<string>>` without having to copy the string.
This for example improves perf of the map_lookup op by 3x.

Differential Revision: [D22830381](https://our.internmc.facebook.com/intern/diff/D22830381/)

[ghstack-poisoned]
